adapters/roundhouseads/roundhouseads.go

```diff
@@ -0,0 +1,196 @@
+package roundhouseads
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+	"text/template"
+
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/macros"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+)
+
+const PREBID_INTEGRATION_TYPE = "1"
+
+type adapter struct {
+	bidderEndpoint string
+}
+
+type ExtImpBidderRoundhouseads struct {
+	adapters.ExtImpBidder
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	pubID, err := getPublisherId(request.Imp)
+
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	modifiedImps := make([]openrtb2.Imp, 0, len(request.Imp))
+
+	for _, imp := range request.Imp {
+
+		if imp.Banner != nil {
+			if len(imp.Banner.Format) > 0 {
+				firstFormat := imp.Banner.Format[0]
+				bannerCopy := *imp.Banner
+				bannerCopy.H = &firstFormat.H
+				bannerCopy.W = &firstFormat.W
+				imp.Banner = &bannerCopy
+
+			}
+		}
+
+		modifiedImps = append(modifiedImps, imp)
+	}
+
+	request.Imp = modifiedImps
+
+	if request.Site != nil {
+		siteCopy := *request.Site
+		if siteCopy.Publisher != nil {
+			publisherCopy := *siteCopy.Publisher
+			if pubID != "" {
+				publisherCopy.ID = pubID
+			}
+			siteCopy.Publisher = &publisherCopy
+		} else {
+			siteCopy.Publisher = &openrtb2.Publisher{ID: pubID}
+		}
+		request.Site = &siteCopy
+	} else if request.App != nil {
+		appCopy := *request.App
+		if appCopy.Publisher != nil {
+			publisherCopy := *appCopy.Publisher
+			if pubID != "" {
+				publisherCopy.ID = pubID
+			}
+			appCopy.Publisher = &publisherCopy
+		} else {
+			appCopy.Publisher = &openrtb2.Publisher{ID: pubID}
+		}
+		request.App = &appCopy
+	}
+
+	errs := make([]error, 0, len(request.Imp))
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-integration-type", PREBID_INTEGRATION_TYPE)
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     a.bidderEndpoint,
+		Body:    reqJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}}, errs
+}
+
+func getPublisherId(impressions []openrtb2.Imp) (string, error) {
+	for _, imp := range impressions {
+
+		var bidderExt ExtImpBidderRoundhouseads
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return "", err
+		}
+
+		var roundhouseadsExt openrtb_ext.ExtImpRoundhouseads
+		if err := json.Unmarshal(bidderExt.Bidder, &roundhouseadsExt); err != nil {
+			return "", err
+		}
+
+		if roundhouseadsExt.PublisherId != "" {
+			return roundhouseadsExt.PublisherId, nil
```

Comment:
Please clearly indicate in your accompanying `prebid.github.io` PR that only the first impression publisher is used and the rest are ignored.